go/app/infra.go

```diff
@@ -35,15 +43,97 @@ func NewItemRepository() ItemRepository {
 
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
-	// STEP 4-2: add an implementation to store an item
+	type ItemsWrapper struct {
+		Items []Item `json:"items"`
+	}
+
+	var wrapper ItemsWrapper
+	data, err := os.ReadFile(i.fileName)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	if err == nil && len(data) > 0 {
+		if err := json.Unmarshal(data, &wrapper); err != nil {
+			return err
+		}
+	}
+
+	wrapper.Items = append(wrapper.Items, Item{Name: item.Name, Category: item.Category, Image: item.Image})
+
+	output, err := json.Marshal(wrapper)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(i.fileName, output, 0644)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+func (i *itemRepository) GetAll(ctx context.Context) (json.RawMessage, error) {
```

Comment:
```suggestion
func (i *itemRepository) GetAll(ctx context.Context) ([]*Item, error) {
```

I think it might be nice to have `GetAll` consistent with `Insert` and `GetByID`!

`GetAll` が `Insert` や `GetByID` と一貫性を持つと良いと思います。